src/codec/rle.rs

```diff
@@ -13,6 +13,7 @@ fn process_plane(
 
     let mut this_line: u32;
     let mut last_line: u32 = 0;
+    let mut raw = 0i8;
```

Comment:
Does it need to be `i8` type?  The first time you set it you read `u8` value into it and cast it to `i8`, and then later when you use it you cast that `i8` type into `u8`. Maybe `u8` type would work from the start and then you could drop those type conversions?

Reply:
Yep, good catch. Fixed.